pages/instructorEffectiveUser/instructorEffectiveUser.js

```diff
@@ -32,6 +33,18 @@ router.post('/', function(req, res, next) {
         res.clearCookie('pl_requested_mode');
         res.redirect(req.originalUrl);
     } else if (req.body.__action == 'changeUid') {
+        if (config.devMode) {
+            // Add the user in devmode to test access rules
+            var params = {
+                uid: req.body.pl_requested_uid,
+                course_instance_id: res.locals.course_instance.id,
+            };
+            return sqldb.query(sql.add_and_enroll, params, function(err) {
```

Comment:
Looks like a race condition here -- we should do the `res.cookie()` and `res.redirect()` inside the callback (which will require duplicating them, I know).

Reply:
Doesn't it already do that (see lines 44, 45)? I don't think I'm following. 

The `return` before sqldb.query ends that branch after the callback is run so lines 48-49 never get run. That's why they're duplicated.

Reply:
You are entirely correct, I was mis-reading the code. Sorry! I'd prefer to change it to remove the `return` on line 42 and instead put 48,49 inside an `else` block, which I find makes the control-flow easier to understand (all paths terminate exactly once in their terminal blocks).